app/services/search/similar_jobs.rb

```diff
@@ -24,9 +24,9 @@ def similar_job_ids
       Search::VacancySearch
         .new(criteria)
         .vacancies
+        .excluding(vacancy)
         .limit(NUMBER_OF_SIMILAR_JOBS * 3) # Fetch more than we need in case some expire while being cached
-        .reject { |job| job.id == vacancy.id }
-        .map(&:id)
+        .pluck(:id)
```

Comment:
Execute everything in DB, saving the instantiation of all the search results to iterate over them.